src/tools/rustfmt/src/config/options.rs

```diff
@@ -513,6 +519,10 @@ pub enum StyleEdition {
     #[doc_hint = "2024"]
     /// [Edition 2024]().
     Edition2024,
+    #[value = "2027"]
+    #[doc_hint = "2027"]
+    /// [Edition 2027]().
+    Edition2027,
```

Comment:
As @compiler-errors mentioned, we'd need to mark this as an unstable option in rustfmt.

```suggestion
    #[value = "2027"]
    #[doc_hint = "2027"]
    #[unstable_variant]
    /// [Edition 2027]().
    Edition2027,
```

Though, If we're not moving forward with the `edition` implementation in this PR, then we can revert these rustfmt changes since we've already got an [unstable 2027 style edition implemented in the upstream rustfmt](https://github.com/rust-lang/rustfmt/blob/a43eef166fa222c32e73608c7f0081603cf47d6a/src/config/options.rs#L517-L521)